force_wfmanager/ui/setup/mco/kpi_specification_view.py

```diff
@@ -100,44 +96,35 @@ def _model_views_default(self):
         """Creates a list of KPISpecificationModelViews for each
         model.kpi"""
         model_views = []
+
         if self.model is not None:
-            # Update model view list
-            model_views += [
-                KPISpecificationModelView(
-                    model=kpi,
-                    available_variables=self.kpi_name_options
-                )
-                for kpi in self.model.kpis
-            ]
+            # Add all MCO KPIs as ModelViews
+            for kpi in self.model.kpis:
+                model_view = KPISpecificationModelView(
+                        model=kpi,
+                        available_variables=self.kpi_name_options
+                    )
+                for var in self.kpi_name_options:
+                    if var.name == kpi.name:
+                        model_view.selected_variable = var
+                model_views.append(model_view)
 
         return model_views
 
     # -------------------
     #     Listeners
     # -------------------
 
-    @cached_property
-    def _get_kpi_names(self):
-        """Listens to model.kpis to extract model names for display"""
-        kpi_names = []
-        for kpi in self.model.kpis:
-            kpi_names.append(kpi.name)
-
-        return kpi_names
-
     @cached_property
     def _get_kpi_name_options(self):
         """Listens to variable_names_registry to extract
          possible names for new KPIs"""
         kpi_name_options = []
         if self.variable_names_registry is not None:
-            outputs = self.variable_names_registry.data_source_outputs
-            inputs = self.variable_names_registry.data_source_inputs
-
-            kpi_name_options += (
-                [output_ for output_ in outputs
-                 if output_ not in inputs]
-            )
+            variables = self.variable_names_registry.available_variables
+            for variable in variables:
+                if len(variable.input_slots) == 0:
```

Comment:
Here we define our criteria for choosing a KPI - in this case no intermediate `Variables` are allowed (those consumed in the execution layer process)